clients/src/test/java/org/apache/kafka/clients/consumer/ConsumerConfigTest.java

```diff
@@ -108,4 +112,42 @@ public void testDefaultPartitionAssignor() {
         assertEquals(Arrays.asList(RangeAssignor.class, CooperativeStickyAssignor.class),
             new ConsumerConfig(properties).getList(ConsumerConfig.PARTITION_ASSIGNMENT_STRATEGY_CONFIG));
     }
+
+    @Test
+    public void testInvalidKeyDeserializer() {
```

Comment:
Maybe also worth covering cases where the values in the `configs` map are `null`, but [this constructor](https://github.com/apache/kafka/blob/6d36487b684fd41522cccd4da4fd88f0b89ff0b7/clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java#L663-L666) is used?

Reply:
Sorry, I didn't quite understand what you meant.
If it refers to the case where the `key.deserializer` value is null, it has been covered.
If it refers to the case where the `value.deserializer` is null, it is reflected in the subsequent testCase `testInvalidValueDeserializer`.

Reply:
I was thinking of some way of simulating people using that constructor in their consumer with a non-null serializer, but a null value for the property for that serializer in the config map. Something like this:

```java
Map<String, Object> props = new HashMap<>();

// populate props with basic required properties like bootstrap.servers

props.put("key.deserializer", null);
props.put("value.deserializer", null);
Consumer<String, String> consumer = new KafkaConsumer<>(props, new StringDeserializer(), new StringDeserializer());
```

It does look like we provide pretty good coverage for that case in the `testAppendDeserializerToConfig` though, so we could also hold off for now. There is technically a gap in coverage with the case where the value for the property is null _and_ the user passes in a null `Deserializer` instance to the consumer, but up to you if you feel it's worth addressing or not.

Reply:
Thanks for your suggestion, I will improve the relevant testcase.